### PR TITLE
support to do benchmark that read binlog from master in any environment

### PR DESCRIPTION
**Pull request by GregoryIan**
### What problem does this PR solve? 
* support to do benchmark that read binlog from master in any environment
* add transform binlog cost monitoring

### What is changed and how it works?
* add a black hole failpoint

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

File: relay/relay.go [hunk 1]
```diff
@@ -364,11 +363,21 @@ func (r *Relay) handleEvents(ctx context.Context, reader2 reader.Reader, transfo
...
+			//r.tctx.L().Info("back hole read binlog takes effects")
+			failpoint.Continue()
+		})
+
+		//r.tctx.L().Info("transform binlog")
 		e := rResult.Event
 		r.tctx.L().Debug("receive binlog event with header", zap.Reflect("header", e.Header))
 
 		// 2. transform events
+		transformimer := time.Now()
```
Comment (csuzhangxc) [inline relay/relay.go]: s/transformimer/transformTimer

File: relay/relay.go [hunk 2]
```diff
@@ -364,11 +363,21 @@ func (r *Relay) handleEvents(ctx context.Context, reader2 reader.Reader, transfo
...
 			return errors.Trace(err)
 		}
+
+		binlogReadDurationHistogram.Observe(time.Since(readTimer).Seconds())
+		failpoint.Inject("BlackholeReadBinlog", func(_ failpoint.Value) {
+			//r.tctx.L().Info("back hole read binlog takes effects")
+			failpoint.Continue()
+		})
+
+		//r.tctx.L().Info("transform binlog")
```
Comment (csuzhangxc) [inline relay/relay.go]: remove these comment lines?

File: relay/metrics.go [hunk 3]
```diff
@@ -110,7 +110,16 @@ var (
...
-			Buckets:   prometheus.ExponentialBuckets(0.00005, 2, 18),
+			Buckets:   prometheus.ExponentialBuckets(0.00005, 2, 20),
+		})
+
+	binlogTransformDurationHistogram = prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Namespace: "dm",
+			Subsystem: "relay",
+			Name:      "read_transform_duration",
+			Help:      "bucketed histogram of read time (s) of transform single binlog event.",
```
Comment (amyangfei) [inline relay/metrics.go]:
```suggestion
			Help:      "bucketed histogram of transform time (s) of single binlog event.",
```

**Pull request merged**